sympy/core/add.py

```diff
@@ -498,7 +498,18 @@ def _eval_derivative(self, s):
         return self.func(*[a.diff(s) for a in self.args])
 
     def _eval_nseries(self, x, n, logx, cdir=0):
-        terms = [t.nseries(x, n=n, logx=logx, cdir=cdir) for t in self.args]
+        from sympy import floor, ceiling
```

Comment:
Special functions should not be imported in core.

Reply:
`ceiling` is imported in `Pow._eval_nseries` too currently.
https://github.com/sympy/sympy/blob/23d11f366de5c112306310313e8aa35b599094f2/sympy/core/power.py#L1494
`floor` is also being imported in evalf.py and expr.py.

I don't think it is possible to capture that the function is `floor` or `ceiling` without importing, nor is it possible to handle the issue without this implementation in `Add`, as `limit(floor(x/2) - floor(x/2 + 1/2), x, oo)` [Any non-integer quantity being added to x causes an issue] must return the unevaluated limit.

I have tried many workarounds before coming to the above conclusion, and I cannot think of a method to proceed with without implementing like this. I would appreciate your help if you know of any alternatives or how to proceed without importing.

Reply:
What is `self` in the case where this is needed.

Reply:
`floor(1/(2*_w)) - floor(0.5 + 1/(2*_w))`

Reply:
@jksuom would it make sense to do something like 
```python
for t in self.args:                                                     
            s = t.nseries(x, n=n, logx=logx, cdir=cdir)                         
            if isinstance(s, Expr):                                             
                check = False                                                   
                for t in self.args:                                             
                    if str(type(t)) == 'floor' or str(type(t)) == 'ceiling':                                 
                        check = True                                            
                        break                                                   
                if check is False:                                              
                    terms.append(s)                                             
                else:                                                           
                    discontinuous.append(s.removeO().subs(x, n))                
            else:                                                               
                terms.append(s)
  ```

Reply:
i think that checking named functions is too specific. There should be a more general way of recognizing the bad cases.

Reply:
What about trying to handle `floor` and `ceiling` separately in limits.py before `gruntz` is called?

I believe there are no special properties which can help here to distinguish `floor(1/x)` to `1/x` when finding the limit about 0, and also to distinguish `sqrt(x) - sqrt(x + 1/2)` (in which both terms evaluate to infinity but sum evaluates to 0) and `floor(1/x) - floor(x + 1/2)` (in which again both terms evaluate to infinity individually but limit is undefined)

Seeing how `ceiling` is included even in the base `Expr.series` (although for a different purpose), and a  I can only see two possible ways to approach this - 
* The current implementation or the string-based one
* Separate handling of `floor` and `ceiling` in limits.py

How do you think this should proceed?